### ISSUE TITLE
Bug: adding a penalty term using `0.1*penalty` fails with error `ERROR: scalar getindex is disallowed`

### ISSUE THREAD

**Issue by xiaodaigh** [FluxML/Zygote.jl#770]
Consider this MWE, actually, I am not 100 sure if the bug is with Zygote or Cuda.jl generated code. 

```
using Flux, CUDA

m = Chain(
    Dense(10, 5, relu),
    Dropout(0.0),
    Dense(5, 5))  |> gpu

CUDA.allowscalar(false)
opt = Flux.Optimise.ADAM()

p = Flux.params(m)

sqnorm(x) = sum(abs2, x)

loss(x,y) = begin
    Flux.Losses.mse(m(x), y) + sum(sqnorm, p)
end

Flux.train!(loss, p, [(rand(10,3), rand(5,3)) |> gpu], opt)
```
the above works, and If I just modify loss a little but multiplying the penalty by 0.1 then it fails with error `ERROR: scalar getindex is disallowed`

```
loss(x,y) = begin
    Flux.Losses.mse(m(x), y) + sum(sqnorm, p)*0.1
end

Flux.train!(loss, p, [(rand(10,3), rand(5,3)) |> gpu], opt)
```

but if I modify it again by dividing by `10` then it works

```
loss(x,y) = begin
    Flux.Losses.mse(m(x), y) + sum(sqnorm, p) / 10
end

Flux.train!(loss, p, [(rand(10,3), rand(5,3)) |> gpu], opt)
```

**Comment by gabrielpreviato** [comment]
Looking at the Stacktrace again, I noted the line `[5] dot(::CuArray{Float64,3}, ::CuArray{Float32,3}) at [...]`.
I think if an operation uses 2 AbstractArrays of different types, the dot product that will be used will do scalar index access (though I'm not sure, I'm new to Julia).

I rewrote the custom loss explicit setting the λ parameter to the Float32 type (using the f0 notation).

```julia
function bbox_loss(ŷ, y, λbbox=0.25f0, agg=sum)
	bbox_w_loss = (ŷ[:, :, 3, :] - y[:, :, 3, :]).^2
	bbox_h_loss = (ŷ[:, :, 4, :] - y[:, :, 4, :]).^2
	
	bbox_loss = (bbox_w_loss + bbox_h_loss) * λbbox
	
	return agg(sum(bbox_loss, dims=(1,2)))
end
```

Now the training works.

@xiaodaigh if you are still having problems, try redefining your loss to
```julia
loss(x,y) = begin
    Flux.Losses.mse(m(x), y) + sum(sqnorm, p) * 0.1f0
end
```
